### PR TITLE
feat: Adds interface to import pipelines

### PR DESCRIPTION
**Pull request by isaacsanders**
I would like to be able to import pipelines/teams. I don't write a lot of Go code nor do I know a lot about the internals of Concourse, so this might not be useful for a number of reasons.

Let me know if there is anything I can do to help land this in the upstream.

### CODE REVIEW

**Comment by isaacsanders** [comment]
@krenair Are you still a maintainer or do you know any maintainers? I'm asking based on you being the one to merge #33

**Comment by isaacsanders** [comment]
@nimalank7 Are you a @alphagov maintainer? Can you help with this PR?

**Comment by isaacsanders** [comment]
@risicle Do you have maintainer on this repo? You are the number 1 committer on https://github.com/terraform-provider-concourse/terraform-provider-concourse/

**Comment by risicle** [comment]
Hi - sorry, missed this. Will try and get some time to look at it at some point.

File: pkg/provider/pipeline.go [hunk 1]
```diff
@@ -415,3 +426,13 @@ func resourcePipelineDelete(d *schema.ResourceData, m interface{}) error {
 	d.SetId("")
 	return nil
 }
+
+// return the pieces of id `left:right` as left, right
+func parseTwoPartID(id, left, right string) (string, string, error) {
+	parts := strings.SplitN(id, ":", 2)
+	if len(parts) != 2 {
```
Comment (risicle) [inline pkg/provider/pipeline.go:433]: Probably worth checking for either of the parts being empty strings too.

FWIW I'm not sure it's worth trying to make this function generic enough to be able to take custom field names for the error messages, it just ends up being confusing what `left` and `right` are actually for.

File: pkg/provider/pipeline.go [hunk 2]
```diff
@@ -12,6 +12,17 @@ import (
 func dataPipeline() *schema.Resource {
 	return &schema.Resource{
 		Read: dataPipelineRead,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```
Comment (risicle) [inline pkg/provider/pipeline.go:16]: According to https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.0.3/helper/schema#ResourceImporter we should be moving towards using `StateContext` here instead, which simply takes an extra `context` argument.

Similarly there's an equivalent `ImportStatePassthroughContext` that can be used with it.

**Comment by risicle** [comment]
Once I've got the integration tests up and working again (see https://github.com/alphagov/terraform-provider-concourse/pull/38), we should add a couple of integration tests to this, and then the only other thing this needs is an addition to the docs covering this.

File: pkg/provider/pipeline.go (hunk 1 above)
Comment (isaacsanders) [reply pkg/provider/pipeline.go:433]: I just copied this from the github provider.

File: pkg/provider/pipeline.go (hunk 2 above)
Comment (isaacsanders) [reply pkg/provider/pipeline.go]: Done

**Comment by isaacsanders** [comment]
@risicle I'm not a Go dev, so the changes to `go.mod` and `go.sum` are not my intention, if you have something better for me to do with them.

**Comment by risicle** [comment]
@isaacsanders are you happy for me to push the contents of the `isaacsanders-feat-import-pipelines` branch to this PR or shall I start another?

https://github.com/alphagov/terraform-provider-concourse/compare/master...isaacsanders-feat-import-pipelines

**Comment by isaacsanders** [comment]
I would have preferred to have the author preserved, but that's fine.

**Comment by isaacsanders** [comment]
Ah, I see. It preserved authorship but changed the committer. This works for me.

**Comment by isaacsanders** [comment]
I've also tested this locally, if you hadn't had a chance.

**Comment by risicle** [comment]
I'm not sure why your avatar didn't show up next to it like it usually does when there's a separate author/committer

**Comment by isaacsanders** [comment]
I would bet it is the use of my work email in the user.

**Comment by isaacsanders** [comment]
I'm happy for this to merge if you are prepared to merge it.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -33,6 +33,8 @@ jobs:
           docker-compose up -d && docker-compose down
 
       - name: integration-tests
+        env:
+          CONCOURSE_VERSION: ${{ matrix.concourse-version }}
         run: |
-          sudo make integration-tests-prep-keys
```
Comment (AP-Hunt) [inline .github/workflows/ci.yml:37]: Tiny ask, but for the sake of quick comprehension could we use the long form of `-E`? `--preserve-env`

Comment (AP-Hunt) [reply .github/workflows/ci.yml:37]: I decided to just do it.

Comment (risicle) [reply .github/workflows/ci.yml:37]: Oh yeah cool sure

**Review by AP-Hunt — APPROVED** [review]
I'm happy with the changes, so I'll approve. I'll make my suggested change in another PR for the sake of expediency.

**Comment by isaacsanders** [comment]
@AP-Hunt What is the process for releasing the feature?

**Comment by AP-Hunt** [comment]
@risicle is taking care of it on our side. 

Rob, are you able to give a timeframe for the release?

**Pull request merged**